### ISSUE TITLE
Could we provide a fallback feature for component "Image"?

### ISSUE THREAD

**Issue by rickyliu521** [diegomura/react-pdf#1810]
There is case that we cannot get the response from src for component "Image", might be 404.  once we have a new fallback feature props, we are able to set default value for it.
In some kind of case, there is no node enviroment for running, we just render PDF by browser.

**Comment by tarabishy2020** [comment]
Would this work?

```jsx
import React from "react";
import { ErrorBoundary as BaseErrorBoundary } from "react-error-boundary";
import { View, StyleSheet, Text } from "@react-pdf/renderer";

const ErrorFallback = ({ error }) => {
  return (
    <View style={styles.View}>
      <Text>Something went wrong:</Text>
      <Text>{error.message}</Text>
    </View>
  );
};
const styles = StyleSheet.create({
  View: {
    width: "100%",
    height: "100%",
    backgroundColor: "red",
    color: "white",
    fontSize: "4pt",
  },
});
export const ErrorBoundary = ({ children, FallbackComponent = null }) => {
  const Fallback = FallbackComponent ? FallbackComponent : ErrorFallback;
  return (
    <BaseErrorBoundary FallbackComponent={Fallback}>
      {children}
    </BaseErrorBoundary>
  );
};
```
use it like this 
```jsx
<ErrorBoundary >
        <ComponentThatCouldError/>
</ErrorBoundary >
```

**Comment by Alarid** [comment]
+1, need this as well. I think the PDF should be generated even if some image requests fail. At the moment, if a single image request results in 404 (or 503, because of the issue described in #1253) the PDF isn't generated, which is unfortunate

**Comment by jeetiss** [comment]
imge fallback can be implemented with current components:

[repl](https://react-pdf-repl.vercel.app/?cp_code=JYWwDg9gTgLgBAbwFBzgMQgOxgGhXAZRgE8AbAUwIAtzzd8ARCAYwFcRzs9UAFAQwDm5bnABqwcgHcRASRCDh-ACrkAHvQC-cAGZQIIOACIAAlHJ9mMALRgAJtoD0ZzLfJmohgNxIkGbADozAWAAZxg3AApkVG0-EGBSYgAuIwB5EMk-UltDERCoZhTDKhgYMBCkhwdtLBgQ_wEwvhhgZn9mfQcQhwgMrNsHADcARgBmBwBNAH0lAGlRADYIAA4AdQhBgCMAVVtRkCmADQBOPgBBM_9S7VykDQBKbyQOzDC4MLJyELgAXkISCjUWgwdpmZrkKL4TYQWzJRD4VBgPi2WzATACFIAJgADCJUAI-GAUsNcfgNHh8LFSKRNhYANZyBQpaKoOCQELAFpYIp8TYhCCkVjhW6suAwCBEuCk0UUbQwFLS1lQYACEoKvFwaGlfTqhFwABeMhcahSVmGInJd0ePheb0ZQlWnKoaCyNPpvzgkNQ-WYIipbuYdIIBQ9xVK5UqDmA_lAAlYUHanRgABYAOJgY5nJT-ABWYAEIrgkmAthgVBENBVJTwD1-AD5PfgADziKTvAHkH4IBBFktlnBwSuq-AaDR1vVN-3kd4FLs-rQOccTqftz5zjv1f20wNTrQ-rtb-nB5gL8eoJsOVuScfWpBqSCwOCuWKsUjwCL3euN89MNgcbBnqyTb8EIq4UOunz1NCsJjnq54qOodasK8rBgA-4S2HAAASACiMgAMJwKACieoegZEd8IRUBAkiYPcF4ITAgGipO8gOk6LrUtudJwayxallQXY4tiGi8agQ4lEJ2IiWJM7MD8YZlBUVTaAkXxxGAFD1B0IBdBpWlRmx5AOJWzB6eAFDDP4pmGLxi4-KK8FqExABSPA4amDHOcxQFTo6ZacQGPGOXxfaCQgwmiSF4nkFWMBSTJ0VyQpJRKZG0axvGia6QASgIFCDFhaCbLm-a2SF9m8U2jF1vhqQ5QQOh8AkpGutxFHvNRtH0Q4NVVX5HFtfSsn8WWCVRdFEnxRF0kTSF-6KRGVSqFY9hWLSITaFYqlaeKUAKFY8hhG4tjNHwNh6LY_ghKM_jkKwVjMJwMD7aQZr-HEfD6lgfCSNpnRTt0qZ6JIZZNEIDgADI5Q4tJQBQxBTJsmzMFQUwAKzHP4YDouVjn2SxDggeQgEXr-7DPTe3hAA&modules=true)

closed as duplicate of #1559
